torch/ao/sparsity/_experimental/activation_sparsifier/README.md

```diff
@@ -0,0 +1,75 @@
+# Activation Sparsifier
+
+## Introduction
+Activation sparsifier attaches itself to a layer(s) in the model and prunes the activations passing through them. **Note that the layer weights are not pruned here.**
+
+## How does it work?
+The idea is to compute a mask to prune the activations. To compute the mask, we need a representative tensor that generalizes activations coming from all the batches in the dataset.
+
+There are 3 main steps involved -
+1. **Aggregation**: The activations coming from inputs across all the batches are aggregated using a user-defined ```aggregate_fn```.
+A simple example is the add function.
+2. **Reduce**: The aggregated activations are then reduced using a user-defined ```reduce_fn```. A simple example is average.
+3. **Masking**: The reduced activations are then passed into a user-defined ```mask_fn``` to compute the mask.
+
+Essentially, the high level idea of computing the mask is
+
+```
+aggregated_tensor = aggregate_fn([activation for activation in all_activations])
+mask = mask_fn(reduce_fn(aggregated_tensor))
+```
+
+*The activation sparsifier also supports per-feature/channel sparsity. This means that a desired set of features in an activation can be also pruned. The mask will be stored per feature.*
+
+```
+for i in range(len(features)):
+    aggregated_tensor_feature = aggregate_fn([activation[features[i]] for activation in all_activations])
+    mask[i] = mask_fn(reduce_fn(aggregated_tensor_feature))
+```
+
+## Implementation Details
+The activation sparsifier attaches itself to a set of layers in a model and then attempts to sparsify the activations flowing through them. *Attach* means registering a [```forward_pre_hook()```](https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#register_forward_pre_hook) to the layer.
```

Comment:
The inline "verbatim" code in markdown uses a single tick (not triple). For example:

**Inline:** "This is an \`inline piece of code\`"  =>  "This is an `inline piece of code`"
**Block:**
\`\`\`
This is a block piece of code
\`\`\`

=> 

```
This is a block piece of code
```